### ISSUE TITLE
[a11y] 11.2.1.1 Keyboard

### ISSUE THREAD

**Issue by tschiebel** [owncloud/android#4366]
audit: https://infinite.owncloud.com/s/QtwuiGHikwFjozV

https://www.etsi.org/deliver/etsi_en/301500_301599/301549/03.02.01_60/en_301549v030201p.pdf
https://github.com/BIK-BITV/BIK-App-Test/tree/main/Pr%C3%BCfschritte/de

### Personal Space 

- [x] If the search at the top is not activated, only the buttons at the top of the page (see screenshot) can be accessed using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/ee189e3a-5b09-4c55-8939-46176440f577)
To access the other interactive elements on the page, you must activate the "X" button (see screenshot). However, you can then return to the top elements of the page, which is a limitation for users who use the keyboard. 
![image](https://github.com/owncloud/android/assets/132444655/c086bf68-f30b-4007-a9df-20fb15dacce8)

- [x] If you tap on a directory or file entry for a long time, a new page opens on which you have the option of selecting directories and files. Unfortunately, this option is not available to keyboard users. Furthermore, it is not immediately apparent to all users that that this option for selecting elements exists.
![image](https://github.com/owncloud/android/assets/132444655/16e24679-01b6-485d-8fe6-31a47ea86309)

### Share

- [x] The "Back" button cannot be reached using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/54d5b120-2782-4e65-b241-a86f68d6fe26)

### Spaces

- [x] If the search at the top is not activated, only the top buttons on the page (see screenshot) can be accessed using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/54c11d40-6ea0-4ec7-b0da-855364f1d9df)
To access the other interactive elements on the page, you must activate the "X" button (see screenshot). However, you can then no longer return to the upper elements of the page, which is a limitation for users who use the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/6860ba14-90cf-4d09-a871-247b6db1698a)

### Manage accounts

- [x] The "Back" button cannot be reached using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/2d75255c-e49e-45f2-9c25-5d5f8d007169)

### Image Viewer

- [x] The buttons at the top of the page cannot be focused or activated using the keyboard. They are invisible to keyboard users and only become visible when tapped on the screen.
![image](https://github.com/owncloud/android/assets/132444655/519d3f1a-0300-48e8-b5d0-1ed8f052c673)

**Comment by jesmrec** [comment]
Try to allign shortcuts in all clients (check iOS)

**Comment by JuancaG05** [comment]
> - If the search at the top is not activated, only the buttons at the top of the page (see screenshot) can be accessed using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/ee189e3a-5b09-4c55-8939-46176440f577)
To access the other interactive elements on the page, you must activate the "X" button (see screenshot). However, you can then return to the top elements of the page, which is a limitation for users who use the keyboard. 
![image](https://github.com/owncloud/android/assets/132444655/c086bf68-f30b-4007-a9df-20fb15dacce8)

Solved with the new keyboard navigation model (explained in https://github.com/owncloud/android/issues/4368#issuecomment-2231256774)

**Comment by JuancaG05** [comment]
> - If you tap on a directory or file entry for a long time, a new page opens on which you have the option of selecting directories and files. Unfortunately, this option is not available to keyboard users. Furthermore, it is not immediately apparent to all users that that this option for selecting elements exists.
![image](https://github.com/owncloud/android/assets/132444655/16e24679-01b6-485d-8fe6-31a47ea86309)

Indeed, it is available for keyboard users as well. They just have to hold `Enter` for a moment (~1 second), just like a long press in touch mode. Then, the toolbar with the different options is reachable as well. Here I upload a video showing it, navigating through the app with a hardware keyboard:

https://github.com/user-attachments/assets/34ceed1a-02df-4c70-81fc-b50fdb53d526

About the gesture, it is one of the main gestures applied in every app in the mobile world. It is a very common UI design pattern, and it even appears as one of the main gestures in Material Design, the official and recommended Google design system for Android apps (which we try to follow as much as possible):
https://m3.material.io/foundations/interaction/gestures#815e151c-d80c-45ab-8720-bc1c2f0191f5

As we can see, it always has the same function: 
> Users can access additional functionality by pressing on elements for an extended time.

Which we follow as well, so this shouldn't be considered a limitation.

**Comment by JuancaG05** [comment]
> - The "Back" button cannot be reached using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/54d5b120-2782-4e65-b241-a86f68d6fe26)

Done. The same action could be performed with the `Esc` key as well.

**Comment by JuancaG05** [comment]
> - The "Back" button cannot be reached using the keyboard.
![image](https://github.com/owncloud/android/assets/132444655/2d75255c-e49e-45f2-9c25-5d5f8d007169)

The "Manage accounts" view has been moved to a dialog. We can exit this new dialog via pressing the "X" button to close it (which is focusable) or by pressing the `Esc` key.

**Comment by JuancaG05** [comment]
> - The buttons at the top of the page cannot be focused or activated using the keyboard. They are invisible to keyboard users and only become visible when tapped on the screen.
![image](https://github.com/owncloud/android/assets/132444655/519d3f1a-0300-48e8-b5d0-1ed8f052c673)

Due to technical restrictions because of the toolbar structure now in the preview activities (image and video), putting the focus on the toolbar (which is an `ActionBar`, a legacy Android API that we'll deprecate at some point **-not in a trivial way-** and that has no support for accessibility common issues, such as focus) is not possible right now. In any case, the actions in the preview toolbar are reachable in some other ways, for example clicking the 3-dot button in the corresponding row in the main file list, or with a long press to the corresponding row in the main file list and selecting the desired option in the toolbar. Since the `Tab` key had no function here, now it is used for the toolbar to appear.
